File: 1ER CORTE/Secante/Secante.js
var grado = parseInt(prompt("Digite grado: "))

var F = []

var tol = parseFloat(prompt("Digite tolerancia: "))

var xo = parseFloat(prompt("Digite xo: "))

var xi = parseFloat(prompt("Digite xi: "))

for(let i = 0; i  <= grado; i++){
    let coeficiente = parseFloat(prompt("Digite coeficiente #"+i+" :"))
    F[i] = coeficiente
}

var sw = 0, it = 0, fxi, fxd, x2 = 0, e

var maxit = parseInt(prompt("Digite número máximo de iteracciones"))
console.log("")
console.log("")
console.log("")

while(sw==0 && it <= maxit){
    fxi = Evaluar(F,xo)
    fxd = Evaluar(F,xi)

    x2 = xi - (((xi-xo)*fxd)/(fxd-fxi))
    e = Math.abs(x2-xi)
    
    console.log("")
    console.log("")
    console.log("Iteracción: " + it 
                +"\nXo     : " + xo
                +"\nXi     : " + xi
                +"\nF(xo)     : " + fxi
                +"\nF(xd)     : " + fxd
                +"\nX2     : " + x2
                +"\nError     : " + e)
    if(e<=tol){
        sw = 1
    }else{
        xo = xi
        xi = x2
        it++
    }
}

if(sw==1){
    console.log("La raíz real es: " + x2)
}else{
    console.log("Not found")
}


function Evaluar(f,x){
    let fx = 0
    for(let i = 0; i < f.length; i++){
        fx+=f[i]*Math.pow(x,i)
    }
    return fx
}

File: 1ER CORTE/Secante/Secante.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
// package RegularFalsi;

// import static Biseccion.Biseccion.Evaluar;
// import java.util.Scanner;

// /**
//  *
//  * @author ASUS
//  */
// public class RegularFalsi {
    
//     public static void main(String[] args) {
        
//         Scanner read = new Scanner(System.in);

//         System.out.println("Digite grado");
//         int grado = read.nextInt();

//         double F[] = new double[grado + 1];

//         System.out.println("Digite tolerancia");
//         double tolerancia = read.nextDouble();

//         for (int i = 0; i < F.length; i++) {
//             System.out.println("Digite coeficiente #" + i);
//             F[i] = read.nextDouble();
//         }
//         int sw = 0;
//         double fxi, fxd;

//         while (sw == 0) {
//             System.out.println("Digite limite inferior");
//             double xi = read.nextDouble();
            
//             System.out.println("Digite limite superior");
//             double xd = read.nextDouble();
            
//             fxi = Evaluar(F, xi);
//             fxd = Evaluar(F, xd);

//             double cer = fxi * fxd;
            
//             if (cer < 0) {
//                 boolean b = true;
//                 double xva = 0, xv,e,fxv;
//                 int it = 1;
//                 while (b) {
                    
//                     xv = xd-((xd-xi)*fxd)/(fxd-fxi);
                    
//                     e = Math.abs(xv-xva);
                  
//                     if (e <= tolerancia) {
//                         System.out.println("\nLa raix real es = " + xv);
//                         b = false;
//                     } else {
//                          fxv = Evaluar(F, xv);
                                                
//                         if (fxv < 0) {
//                             xi = xv;
//                         } else {
//                             xd = xv;
//                         }
//                         System.out.println(xv);
//                     }
//                     it++;
//                     xva = xv;
//                 }
//                 sw = 1;
//             }else{
//                 System.out.println("No existe una raix entre estos inventarios");
//                 System.out.println("fxi = " + fxi + "\nfxd = " + fxd);
//             }
//         }
        
//     }
//     public static double Evaluar(double f[], double x) {
//         double fx = 0;
//         for (int i = 0; i < f.length; i++) {
//             fx += f[i] * Math.pow(x, i);
//         }
//         return fx;
//     }
    
// }
